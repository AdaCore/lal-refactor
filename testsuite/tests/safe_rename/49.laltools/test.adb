package body Test is
   function Increment (Value : Integer) return Integer is
   begin
      return Value + 1;
   end Increment;

   procedure Bar is
      Number : constant Integer :=
        Increment (1);
   begin
         null;
   end Bar;
end Test;
