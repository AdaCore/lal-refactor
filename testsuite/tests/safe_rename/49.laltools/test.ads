package Test is
   function Increment (Value : Integer) return Integer;
end Test;
