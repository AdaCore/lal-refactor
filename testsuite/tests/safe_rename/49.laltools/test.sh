safe_rename -P test.gpr -S test.adb -L 4 -R 16 -N Number --algorithm analyse_ast
