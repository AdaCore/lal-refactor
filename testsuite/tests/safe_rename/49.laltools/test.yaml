description: |
    Refactoring Safe Rename test

    Rename the `Value` parameter of the function `Increment` to `Number`,
    which has the same name as a local variable in procedure `Bar`, where
    `Increment` is used. Confirm that no collision is detected.

driver: shell_script
