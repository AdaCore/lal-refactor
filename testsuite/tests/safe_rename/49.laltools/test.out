# Renaming node test.adb:4:14: 
References:
Unit:test.adb
Line: 2
Source Location Range:
Start_Line: 2
Start_Column: 24
End_Line: 2
End_Column: 29
Line: 4
Source Location Range:
Start_Line: 4
Start_Column: 14
End_Line: 4
End_Column: 19
Unit:test.ads
Line: 2
Source Location Range:
Start_Line: 2
Start_Column: 24
End_Line: 2
End_Column: 29

